### PR TITLE
Add `__repr__` and `_repr_html_` to `SequenceCollection`

### PR DESCRIPTION
**Pull request by rmcar17**
Adds `__repr__` and `_repr_html_` to `SequenceCollection` resolving #1682.

`__repr__` has been tested. `_repr_html_` has only been tested through the existing (slightly modified) `test_set_wrap_affects_repr_html` tests. Further explicit testing may be required in `SequenceCollectionTests`

### CODE REVIEW

File: tests/test_core/test_alignment.py [hunk 1]
```diff
@@ -985,6 +981,56 @@ def test_construction(self):
         seq2 = make_seq("AC", name="seq2")
         coll = SequenceCollection(data=[seq1, seq2])
 
+    def test_repr(self):
```
Comment (GavinHuttley) [inline tests/test_core/test_alignment.py:984]: We are gradually converting existing tests to pytest style. Can you cut this method at the bottom of the file and edit as follows
```python
@pytest.mark.parametrize("cls", [SequenceCollection, Alignment, ArrayAlignment])
def test_repr(cls):
    ...
    seqs = cls(data=data, moltype=DNA)
    # replace all self.assertEqual calls as per
    assert repr(seqs) == "2x (ENSMUSG00000039616[GCCCTTCAAAT...], ENSMUSG00000056468[GCCAGGGGGGA...]) <dna> seqcollection"
```
etc...

File: src/cogent3/core/alignment.py [hunk 2]
```diff
@@ -2144,6 +2144,202 @@ def distance_matrix(self, calc="pdist"):
...
+            if len(elts) > limit:
+                elts.append("...")
+            seqs.append(f"{name}[{delimiter.join(elts)}]")
+
+        if len(self.names) > 2:
+            seqs.insert(1, "...")
+
+        seqs = ", ".join(seqs)
+
+        return f"{len(self.names)}x ({seqs}) <{self.moltype.get_type()}> seqcollection"
```
Comment (GavinHuttley) [inline src/cogent3/core/alignment.py:2174]: remove angle brackets around the moltype.

File: tests/test_core/test_alignment.py [hunk 3]
```diff
@@ -985,6 +981,56 @@ def test_construction(self):
...
+        data = {
+            "ENSMUSG00000056468": "GCCAGGGGGGAAAGGGAGAA",
+            "ENSMUSG00000039616": "GCCCTTCAAATTT",
+        }
+        seqs = self.Class(data=data, moltype=DNA)
+        self.assertEqual(
+            repr(seqs),
+            "2x (ENSMUSG00000039616[GCCCTTCAAAT...], ENSMUSG00000056468[GCCAGGGGGGA...]) <dna> seqcollection",
```
Comment (GavinHuttley) [inline tests/test_core/test_alignment.py:992]: Remove angle brackets

**Review by GavinHuttley — CHANGES_REQUESTED** [review]
awesome, some small changes and done!

File: tests/test_core/test_alignment.py (hunk 1 above)
Comment (rmcar17) [reply tests/test_core/test_alignment.py:984]: This was written in only the `SequenceCollectionTests` class so I've resolved this without parametrizing, output for `Alignment` and `ArrayAlignment` should be different

**Review by GavinHuttley — APPROVED** [review]
Awesome, thanks @rmcar17 !

**Pull request merged**